### DISCUSSION TITLE
Detect animated WebP

### DISCUSSION THREAD

**Discussion by nevaran** [dlemstra/Magick.NET#814] [General]
Is there any way to find if webp files are animated or not?
And can I load just the first frame if it has multiple and save on some processing time?

**Comment by dlemstra** [comment]
You could use the `MagickImageInfo` class to get the number of frames:

```C#
var count = MagickImageInfo.ReadCollection("yourFile.webp").Count();
```

And if you only want the first frame you can do this:

```C#
using (var image = new MagickImage("yourFile.webp"))
{
}
```

**Reply by nevaran** [reply]
Ah I see; Does it also work for png, jpg, and any other image types that can have animation?

**Reply by dlemstra** [reply]
This also works for other formats.

p.s. jpg cannot be an animation.

**Reply by nevaran** [reply]
Yes sorry, I remembered a jpg that was an animation; but it turned out it was just changed the extension type from a gif haha

**Reply by nevaran** [reply]
Are you able to get width/height by using `ReadCollection` ?

**Reply by dlemstra** [reply]
You will get basic information of the image. The width and height are included in this.

**Reply by nevaran** [reply]
I just tried it out, and apparently it does not seem to work for gifs that have their extensions changed to jpg, for example. The returned count is always 1
Edit: Never mind, it was my code's fault, though it does not seem to detect animated png
